src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -810,31 +811,62 @@ private LanguageExpression ConvertBinary(BinaryOperationSyntax syntax)
 
         private LanguageExpression ConvertUnary(UnaryOperationSyntax syntax)
         {
-            LanguageExpression convertedOperand = ConvertExpression(syntax.Expression);
-
             switch (syntax.Operator)
             {
                 case UnaryOperator.Not:
+                    LanguageExpression convertedOperand = ConvertExpression(syntax.Expression);
                     return CreateFunction("not", convertedOperand);
 
                 case UnaryOperator.Minus:
-                    if (convertedOperand is JTokenExpression literal && literal.Value.Type == JTokenType.Integer)
+                    if (syntax.Expression is IntegerLiteralSyntax integerLiteral)
                     {
-                        // invert the integer literal
-                        int literalValue = literal.Value.Value<int>();
-                        return new JTokenExpression(-literalValue);
+                        //do codegen logic
+                        LanguageExpression integerExpression = ConvertInteger(integerLiteral, true);
+                        return integerExpression;
                     }
 
                     return CreateFunction(
                         "sub",
                         new JTokenExpression(0),
-                        convertedOperand);
+                        ConvertExpression(syntax.Expression));
 
                 default:
                     throw new NotImplementedException($"Cannot emit unexpected unary operator '{syntax.Operator}.");
             }
         }
 
+        // for 32 bit integers, we return the integer value
+        // for values outside that range, return the FunctionExpression
+        private LanguageExpression ConvertInteger(IntegerLiteralSyntax integerSyntax, bool minus)
+        {
+            long integerValue = (long)integerSyntax.Value;
```

Comment:
This feels like something to be careful with, as you haven't checked that `integerSyntax.Value` is inside the bounds of a `long`.

I think you could probably just avoid this cast by:
1. Avoiding performing math on the value in the negative case (lines 850 - 851) - you could instead just do: `new JTokenExpression($"-{integerSyntax.Value.ToInvariantString()}")`
1. Use the value directly in the positive case (line 861) - you could instead just do: `new JTokenExpression(integerSyntax.Value.ToInvariantString())`